### PR TITLE
Use webdriver to set permission on getUserMedia calls 

### PR DESCRIPTION
**Pull request by dontcallmedom**
this includes removing the default fake-ui from chrome wpt runner - this makes obtaining permission a prerequisite for using `getUserMedia` in tests.

### CODE REVIEW

**Review by reillyeon — APPROVED** [review]
Overall this change seems reasonable but I only took a close look at `mediacapture-image/ImageCapture-creation.https.html` since that's the file I'm an owner for.

File: mediacapture-record/MediaRecorder-bitrate.https.html [hunk 1]
```diff
@@ -5,6 +5,9 @@
 <link rel="help" href="https://w3c.github.io/mediacapture-record/MediaRecorder.html">
 <script src="/resources/testharness.js"></script>
 <script src="/resources/testharnessreport.js"></script>
+<script src=/resources/testdriver.js></script>
+<script src=/resources/testdriver-vendor.js></script>
+<script src='../mediacapture-streams/permission-helper.js'></script>
```
Comment (alvestrand) [inline mediacapture-record/MediaRecorder-bitrate.https.html:10]: can we please have one convention on how filenames are quoted in "script src=" statements?
This makes 3 different versions in a single file.

File: mediacapture-streams/MediaStream-MediaElement-preload-none.https.html [hunk 2]
```diff
@@ -0,0 +1,83 @@
+<!DOCTYPE html>
+<html>
+    <head>
+        <title>Test that the HTMLMediaElement preload 'none' attribute value is ignored for MediaStream used as srcObject and MediaStream object URLs used as src.</title>
+        <link rel="author" title="Matthew Wolenetz" href="mailto:wolenetz@chromium.org"/>
```
Comment (alvestrand) [inline mediacapture-streams/MediaStream-MediaElement-preload-none.https.html:5]: This file seems like it belongs to a different PR.

File: mediacapture-streams/permission-helper.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+async function setPermission(status="granted", scope=["camera", "microphone"]) {
+  try {
+    for (let s of scope) {
+      await test_driver.set_permission({ name: s }, status, true);
+    }
+  } catch {
```
Comment (alvestrand) [inline mediacapture-streams/permission-helper.js:6]: generic catch is usually a bad idea. What error do we expect here?

File: mediacapture-streams/permission-helper.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+async function setPermission(status="granted", scope=["camera", "microphone"]) {
```
Comment (alvestrand) [inline mediacapture-streams/permission-helper.js:1]: the "status" argument is never used. Is it a future goal to be able to set setPermission(status=denied, scope=["camera"])? If so, document (or even better, implement).

**Comment by alvestrand** [comment]
So the harness-setting-permission function has finally landed? Great!

A couple of things:

- the name setPermission() is a bad idea. setMediaDevicePermission() is longer, but won't conflict with everyone else's helpers for setting their particular permissions. When we go to tests that mix mulitple interfaces with separate permissions they want to set, it's important that they don't get to clobber each other.

- setting permission for camera when microphone only is needed by the test is sloppy and will fail to detect errors involving checking the wrong permission

- this is a murderous lot of file changes for doing the same thing in multiple places. Would it be possible to wrap this inside a helper function rather than repeating the same 3 lines over and over again?

- I can't see that this PR actually removes the Chrome flag. So the tests will pass on Chrome whether or not the test driver function exists or not. Is this intentional, or is it "delayed to the next PR"?
-

File: mediacapture-streams/MediaStream-MediaElement-preload-none.https.html (hunk 2 above)
Comment (dontcallmedom) [reply mediacapture-streams/MediaStream-MediaElement-preload-none.https.html:5]: that file was a `-manual` file, presumably because when it was created, one would not gain access to getUserMedia. Now that we do, it feels reasonable to make the move here.

File: mediacapture-streams/permission-helper.js (hunk 3 above)
Comment (dontcallmedom) [reply mediacapture-streams/permission-helper.js:6]: this is to deal with browsers that don't implement the `set_permission` webdriver command (i.e. both Firefox and Safari at this time point).

File: mediacapture-streams/permission-helper.js (hunk 4 above)
Comment (dontcallmedom) [reply mediacapture-streams/permission-helper.js:1]: it's used in `GUM-deny.html`

**Comment by dontcallmedom** [comment]
>     * the name setPermission() is a bad idea. setMediaDevicePermission() is longer, but won't conflict with everyone else's helpers for setting their particular permissions. When we go to tests that mix mulitple interfaces with separate permissions they want to set, it's important that they don't get to clobber each other.

Will fix to `setMediaDevicePermission()`
 
>     * setting permission for camera when microphone only is needed by the test is sloppy and will fail to detect errors involving checking the wrong permission

I can take a pass at setting permissions more specifically.

>     * this is a murderous lot of file changes for doing the same thing in multiple places. Would it be possible to wrap this inside a helper function rather than repeating the same 3 lines over and over again?

I sure wish it was; but doing this cleanly would require using JS modules, which I don't think the WPT infrastructure is ready for at this point.

>     * I can't see that this PR actually removes the Chrome flag. So the tests will pass on Chrome whether or not the test driver function exists or not. Is this intentional, or is it "delayed to the next PR"?

It does remove the flag, see https://github.com/web-platform-tests/wpt/pull/31847/files#diff-0df24b5b583c460182e687f7dc7a6a79dd2cd3389bc4a96f48483f60fceb51f7

File: mediacapture-streams/permission-helper.js (hunk 3 above)
Comment (alvestrand) [reply mediacapture-streams/permission-helper.js:6]: so we should catch "no such function". I'm worried about the case where it fails for an unrelated reason (such as calling with "scope=['cammera']" and getting a "no such value" exception - that could be hard to debug when the error is caught and ignored).

**Review by alvestrand — APPROVED** [review]
This is a great step forward in testing permission-gated functionality!
